packages/carousel/src/Carousel.js

```diff
@@ -22,17 +21,35 @@ export const Carousel = ({
   orientation = 'horizontal',
   naturalSlideWidth = 100,
   visibleSlides = 1,
+  arrowsPosition = 'side',
 }) => {
+  const formattedVisibleSlides = getVisibleSlidesArray(visibleSlides)
   const widths = layoutToFlexWidths(layout, children.length)
   const layoutSize = layout?.split('-').length
 
+  const useResponsiveVisibleSlides = (visibleSlides) => {
+    const [windowWidth, setWindowWidth] = useState(window.innerWidth)
+    useEffect(() => {
+      const handleResize = () => {
+        setWindowWidth(window.innerWidth)
+      }
+      window.addEventListener('resize', handleResize)
```

Comment:
Should this be debounced at all?

Reply:
Might be a good idea - were you thinking `lodash` debounce?

Reply:
we could separate this hook out to another file to make it easier to read/test

Reply:
I originally had this in a separate file, but I was getting a code scan error saying it must be defined inside a component

Reply:
I was able to pull this file out and pass the scan, it just has to have `use` to begin the function name

Reply:
nice! yeah that makes sense